react/react1/week2/todo-list/src/AddTodo.jsx

```diff
@@ -0,0 +1,31 @@
+import React,{ useState}  from 'react';
+
+import TodoList from "./List";
+import todoList  from './data';
+<todoList todoList = {todoList}/>
+
+export default function AddTodo () { 
+  const  [list,setList] = useState(todoList);
+
+  const addTaskClick= ()=> {
+    const arryLen = list.length;
+    const newTodo = `Random Todo has been Added`;
+    //const newTodoList = list.concat([{ id: '' + (arryLen + 1) , description: `${newTodo}`}]);
+    setList([...list, {id:'' + (arryLen + 1) , description: `${newTodo}`}])
```

Comment:
Your id needs to be exclusive, also because you're using it as the index, so make it invariable. You can use [Date.now()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now), for example, bc it will never be the same.